tracer/src/Datadog.Trace/Propagators/DatadogContextPropagator.cs

```diff
@@ -58,8 +67,12 @@ samplingPriority.Value switch
             var parentId = ParseUtility.ParseUInt64(carrier, carrierGetter, HttpHeaderNames.ParentId) ?? 0;
             var samplingPriority = ParseUtility.ParseInt32(carrier, carrierGetter, HttpHeaderNames.SamplingPriority);
             var origin = ParseUtility.ParseString(carrier, carrierGetter, HttpHeaderNames.Origin);
+            var propagatedTraceTags = ParseUtility.ParseString(carrier, carrierGetter, HttpHeaderNames.PropagatedTags);
 
-            spanContext = new SpanContext(traceId, parentId, samplingPriority, serviceName: null, origin);
+            spanContext = new SpanContext(traceId, parentId, samplingPriority, serviceName: null, origin)
+                          {
+                              PropagatedTags = propagatedTraceTags
+                          };
```

Comment:
Back when I added `DatadogTags` (renamed to `PropagatedTags` in this PR) I was trying to avoid adding more ctor parameters. As we keep adding more and more values to the span context, that will not scale well.

But I think you're right. I'll change it for now and hopefully we can come back and refactor all the parameters later.

Reply:
On second thought, I want to keep this as a property setter. I am setting `PropagatedTags` in two places that use different constructor overloads, and `SpanContext` already has too many confusing constructor overloads. Adding this parameter to both would add even more complexity.

I do want to come back at some time later and refactor `SpanContext` and clean up all the ctors (and stop using `SpanContext` entirely when not needed).

Reply:
I don't completely agree with the complexity of the .ctor. For me it's a worse complexity to have a setter for something that should be set at construction only. 
As it's all internal and as we're on a tight schedule, I don't mind coming back to it later though indeed.